### ISSUE TITLE
Error: key does not exist, body not found

### ISSUE THREAD

**Issue by alicelogos** [restic/restic#4752]
Output of `restic version`
--------------------------

```
restic 0.16.4 compiled with go1.22.1 on linux/amd64
```

What backend/service did you use to store the repository?
---------------------------------------------------------

rclone, Microsoft OneDrive

Problem description / Steps to reproduce
----------------------------------------

Any command doesn't work with my repository. For example,

```
restic --repo rclone:***myrepo*** snapshots
```

Command line output:

```
debug log file /tmp/restic-debug.log
debug enabled
enter password for repository:
body not drained, 10 bytes not read, error: unexpected EOF, body: "Not Found\n"
Load(<key/f221949399>, 0, 0) returned error, retrying after 622.465913ms: <key/f221949399> does not exist
body not drained, 10 bytes not read, error: unexpected EOF, body: "Not Found\n"
Load(<key/f221949399>, 0, 0) returned error, retrying after 628.461006ms: <key/f221949399> does not exist
body not drained, 10 bytes not read, error: unexpected EOF, body: "Not Found\n"
Load(<key/f221949399>, 0, 0) returned error, retrying after 940.196547ms: <key/f221949399> does not exist
body not drained, 10 bytes not read, error: unexpected EOF, body: "Not Found\n"
Load(<key/f221949399>, 0, 0) returned error, retrying after 1.044828737s: <key/f221949399> does not exist
body not drained, 10 bytes not read, error: unexpected EOF, body: "Not Found\n"
Load(<key/f221949399>, 0, 0) returned error, retrying after 3.761432006s: <key/f221949399> does not exist
body not drained, 10 bytes not read, error: unexpected EOF, body: "Not Found\n"
Load(<key/f221949399>, 0, 0) returned error, retrying after 2.67632065s: <key/f221949399> does not exist
body not drained, 10 bytes not read, error: unexpected EOF, body: "Not Found\n"
Load(<key/f221949399>, 0, 0) returned error, retrying after 4.534459853s: <key/f221949399> does not exist
body not drained, 10 bytes not read, error: unexpected EOF, body: "Not Found\n"
Load(<key/f221949399>, 0, 0) returned error, retrying after 12.186197623s: <key/f221949399> does not exist
body not drained, 10 bytes not read, error: unexpected EOF, body: "Not Found\n"
Load(<key/f221949399>, 0, 0) returned error, retrying after 12.450027669s: <key/f221949399> does not exist
body not drained, 10 bytes not read, error: unexpected EOF, body: "Not Found\n"
Load(<key/f221949399>, 0, 0) returned error, retrying after 9.778842663s: <key/f221949399> does not exist
body not drained, 10 bytes not read, error: unexpected EOF, body: "Not Found\n"
<key/f221949399> does not exist. Try again
                               signal interrupt received, cleaning up

```

Content of `/tmp/restic-debug.log`:

```
2024/04/04 09:59:12 restic/main.go:106	main.main	1	main []string{"/etc/profiles/per-user/***myusername***/bin/restic", "--repo", "rclone:***myrepo***", "snapshots"}
2024/04/04 09:59:12 restic/main.go:107	main.main	1	restic 0.16.4 compiled with go1.22.1 on linux/amd64
2024/04/04 09:59:12 restic/global.go:575	main.open	1	parsing location rclone:***myrepo***
2024/04/04 09:59:12 restic/global.go:569	main.parseConfig	1	opening rclone repository at &rclone.Config{Program:"rclone", Args:"serve restic --stdio --b2-hard-delete", Remote:"***myrepo***", Connections:0x5, Timeout:60000000000}
2024/04/04 09:59:12 rclone/backend.go:171	rclone.newBackend	1	running command: rclone [serve restic --stdio --b2-hard-delete ***myrepo***]
2024/04/04 09:59:12 debug/round_tripper.go:93	debug.loggingRoundTripper.RoundTrip	1	------------  HTTP REQUEST -----------
GET /file-275122350061446781 HTTP/1.1
Host: localhost
User-Agent: Go-http-client/1.1
Accept: application/vnd.x.restic.rest.v2
Accept-Encoding: gzip


2024/04/04 09:59:12 rclone/backend.go:186	rclone.newBackend.func1	83	new connection requested, tcp localhost:80
2024/04/04 09:59:15 debug/round_tripper.go:110	debug.loggingRoundTripper.RoundTrip	1	------------  HTTP RESPONSE ----------
HTTP/2.0 404 Not Found
Content-Length: 10
Accept-Ranges: bytes
Content-Type: text/plain; charset=utf-8
Date: Thu, 04 Apr 2024 01:59:15 GMT
Server: rclone/1.66.0
X-Content-Type-Options: nosniff


2024/04/04 09:59:15 rclone/backend.go:254	rclone.newBackend	1	HTTP status "404 Not Found" returned, moving instance to background
2024/04/04 09:59:15 logger/log.go:52	logger.(*Backend).Stat	1	Stat(<config/>)
2024/04/04 09:59:15 sema/semaphore.go:27	sema.semaphore.GetToken	1	acquired token
2024/04/04 09:59:15 debug/round_tripper.go:93	debug.loggingRoundTripper.RoundTrip	1	------------  HTTP REQUEST -----------
HEAD /config HTTP/1.1
Host: localhost
User-Agent: Go-http-client/1.1
Accept: application/vnd.x.restic.rest.v2


2024/04/04 09:59:16 debug/round_tripper.go:110	debug.loggingRoundTripper.RoundTrip	1	------------  HTTP RESPONSE ----------
HTTP/2.0 200 OK
Content-Length: 155
Accept-Ranges: bytes
Date: Thu, 04 Apr 2024 01:59:16 GMT
Last-Modified: Tue, 04 Jul 2023 14:40:37 GMT
Server: rclone/1.66.0


2024/04/04 09:59:16 logger/log.go:54	logger.(*Backend).Stat	1	  stat err <nil>
2024/04/04 09:59:16 index/index.go:382	index.(*Index).Finalize	1	finalizing index
2024/04/04 09:59:21 logger/log.go:59	logger.(*Backend).List	1	List(key)
2024/04/04 09:59:21 debug/round_tripper.go:93	debug.loggingRoundTripper.RoundTrip	1	------------  HTTP REQUEST -----------
GET /keys/ HTTP/1.1
Host: localhost
User-Agent: Go-http-client/1.1
Accept: application/vnd.x.restic.rest.v2
Accept-Encoding: gzip


2024/04/04 09:59:22 debug/round_tripper.go:110	debug.loggingRoundTripper.RoundTrip	1	------------  HTTP RESPONSE ----------
HTTP/2.0 200 OK
Content-Length: 89
Accept-Ranges: bytes
Content-Type: application/vnd.x.restic.rest.v2
Date: Thu, 04 Apr 2024 01:59:22 GMT
Server: rclone/1.66.0


2024/04/04 09:59:22 rest/rest.go:389	rest.(*Backend).listv2	1	parsing API v2 response
2024/04/04 09:59:22 repository/key.go:145	repository.SearchKey.func1	1	trying key "f221949399db1b03be9a182fb361cedf4e3ea96190f6ec030c8131cfee053220"
2024/04/04 09:59:22 logger/log.go:45	logger.(*Backend).Load	1	Load(<key/f221949399>, length 0, offset 0)
2024/04/04 09:59:22 sema/semaphore.go:27	sema.semaphore.GetToken	1	acquired token
2024/04/04 09:59:22 debug/round_tripper.go:93	debug.loggingRoundTripper.RoundTrip	1	------------  HTTP REQUEST -----------
GET /keys/f221949399db1b03be9a182fb361cedf4e3ea96190f6ec030c8131cfee053220 HTTP/1.1
Host: localhost
User-Agent: Go-http-client/1.1
Accept: application/vnd.x.restic.rest.v2
Range: bytes=0-


2024/04/04 09:59:25 debug/round_tripper.go:110	debug.loggingRoundTripper.RoundTrip	1	------------  HTTP RESPONSE ----------
HTTP/2.0 404 Not Found
Content-Length: 447
Accept-Ranges: bytes
Content-Range: bytes 0-446/447
Content-Type: text/plain; charset=utf-8
Date: Thu, 04 Apr 2024 01:59:25 GMT
Last-Modified: Tue, 04 Jul 2023 14:40:29 GMT
Server: rclone/1.66.0
X-Content-Type-Options: nosniff


2024/04/04 09:59:25 debug/round_tripper.go:46	debug.(*eofDetectReader).Close	1	body not drained, 10 bytes not read, error: unexpected EOF, body: "Not Found\n": Close()
github.com/restic/restic/internal/debug.(*eofDetectReader).Close
	github.com/restic/restic/internal/debug/round_tripper.go:46
github.com/restic/restic/internal/backend/rest.(*Backend).openReader
	github.com/restic/restic/internal/backend/rest/rest.go:228
github.com/restic/restic/internal/backend/rest.(*Backend).Load
	github.com/restic/restic/internal/backend/rest/rest.go:176
github.com/restic/restic/internal/backend/sema.(*connectionLimitedBackend).Load
	github.com/restic/restic/internal/backend/sema/backend.go:95
github.com/restic/restic/internal/backend/logger.(*Backend).Load
	github.com/restic/restic/internal/backend/logger/log.go:46
github.com/restic/restic/internal/backend/retry.(*Backend).Load.func1
	github.com/restic/restic/internal/backend/retry/backend_retry.go:131
github.com/restic/restic/internal/backend/retry.retryNotifyErrorWithSuccess.func1
	github.com/restic/restic/internal/backend/retry/backend_retry.go:47
github.com/cenkalti/backoff/v4.RetryNotifyWithTimer.Operation.withEmptyData.func1
	github.com/cenkalti/backoff/v4@v4.2.1/retry.go:18
github.com/cenkalti/backoff/v4.doRetryNotify[...]
	github.com/cenkalti/backoff/v4@v4.2.1/retry.go:88
github.com/cenkalti/backoff/v4.RetryNotifyWithTimer
	github.com/cenkalti/backoff/v4@v4.2.1/retry.go:61
github.com/cenkalti/backoff/v4.RetryNotify
	github.com/cenkalti/backoff/v4@v4.2.1/retry.go:49
github.com/restic/restic/internal/backend/retry.retryNotifyErrorWithSuccess
	github.com/restic/restic/internal/backend/retry/backend_retry.go:55
github.com/restic/restic/internal/backend/retry.(*Backend).retry
	github.com/restic/restic/internal/backend/retry/backend_retry.go:77
github.com/restic/restic/internal/backend/retry.(*Backend).Load
	github.com/restic/restic/internal/backend/retry/backend_retry.go:129
github.com/restic/restic/internal/backend.LoadAll
	github.com/restic/restic/internal/backend/utils.go:19
github.com/restic/restic/internal/repository.LoadKey
	github.com/restic/restic/internal/repository/key.go:182
github.com/restic/restic/internal/repository.OpenKey
	github.com/restic/restic/internal/repository/key.go:66
github.com/restic/restic/internal/repository.SearchKey.func1
	github.com/restic/restic/internal/repository/key.go:146
github.com/restic/restic/internal/repository.SearchKey.(*Repository).List.func2
	github.com/restic/restic/internal/repository/repository.go:887
github.com/restic/restic/internal/backend/retry.(*Backend).List.func1.1
	github.com/restic/restic/internal/backend/retry/backend_retry.go:178
github.com/restic/restic/internal/backend/rest.(*Backend).listv2
	github.com/restic/restic/internal/backend/rest/rest.go:410
github.com/restic/restic/internal/backend/rest.(*Backend).List
	github.com/restic/restic/internal/backend/rest/rest.go:345
github.com/restic/restic/internal/backend/logger.(*Backend).List
	github.com/restic/restic/internal/backend/logger/log.go:60
github.com/restic/restic/internal/backend/retry.(*Backend).List.func1
	github.com/restic/restic/internal/backend/retry/backend_retry.go:172
github.com/restic/restic/internal/backend/retry.retryNotifyErrorWithSuccess.func1
	github.com/restic/restic/internal/backend/retry/backend_retry.go:47
github.com/cenkalti/backoff/v4.RetryNotifyWithTimer.Operation.withEmptyData.func1
	github.com/cenkalti/backoff/v4@v4.2.1/retry.go:18
github.com/cenkalti/backoff/v4.doRetryNotify[...]
	github.com/cenkalti/backoff/v4@v4.2.1/retry.go:88
github.com/cenkalti/backoff/v4.RetryNotifyWithTimer
	github.com/cenkalti/backoff/v4@v4.2.1/retry.go:61
github.com/cenkalti/backoff/v4.RetryNotify
	github.com/cenkalti/backoff/v4@v4.2.1/retry.go:49
github.com/restic/restic/internal/backend/retry.retryNotifyErrorWithSuccess
	github.com/restic/restic/internal/backend/retry/backend_retry.go:55
github.com/restic/restic/internal/backend/retry.(*Backend).retry
	github.com/restic/restic/internal/backend/retry/backend_retry.go:77
github.com/restic/restic/internal/backend/retry.(*Backend).List
	github.com/restic/restic/internal/backend/retry/backend_retry.go:171
2024/04/04 09:59:25 logger/log.go:47	logger.(*Backend).Load	1	  load err <key/f221949399> does not exist
2024/04/04 09:59:25 restic/global.go:290	main.Warnf	1	Load(<key/f221949399>, 0, 0) returned error, retrying after 622.465913ms: <key/f221949399> does not exist
2024/04/04 09:59:26 logger/log.go:45	logger.(*Backend).Load	1	Load(<key/f221949399>, length 0, offset 0)
2024/04/04 09:59:26 sema/semaphore.go:27	sema.semaphore.GetToken	1	acquired token
2024/04/04 09:59:26 debug/round_tripper.go:93	debug.loggingRoundTripper.RoundTrip	1	------------  HTTP REQUEST -----------
GET /keys/f221949399db1b03be9a182fb361cedf4e3ea96190f6ec030c8131cfee053220 HTTP/1.1
Host: localhost
User-Agent: Go-http-client/1.1
Accept: application/vnd.x.restic.rest.v2
Range: bytes=0-



...... (repeat debug/round_tripper.go:110 and debug/round_tripper.go:46)



2024/04/04 10:00:29 debug/round_tripper.go:110	debug.loggingRoundTripper.RoundTrip	1	------------  HTTP RESPONSE ----------
HTTP/2.0 404 Not Found
Content-Length: 447
Accept-Ranges: bytes
Content-Range: bytes 0-446/447
Content-Type: text/plain; charset=utf-8
Date: Thu, 04 Apr 2024 02:00:29 GMT
Last-Modified: Tue, 04 Jul 2023 14:40:29 GMT
Server: rclone/1.66.0
X-Content-Type-Options: nosniff


2024/04/04 10:00:29 debug/round_tripper.go:46	debug.(*eofDetectReader).Close	1	body not drained, 10 bytes not read, error: unexpected EOF, body: "Not Found\n": Close()
github.com/restic/restic/internal/debug.(*eofDetectReader).Close
	github.com/restic/restic/internal/debug/round_tripper.go:46
github.com/restic/restic/internal/backend/rest.(*Backend).openReader
	github.com/restic/restic/internal/backend/rest/rest.go:228
github.com/restic/restic/internal/backend/rest.(*Backend).Load
	github.com/restic/restic/internal/backend/rest/rest.go:176
github.com/restic/restic/internal/backend/sema.(*connectionLimitedBackend).Load
	github.com/restic/restic/internal/backend/sema/backend.go:95
github.com/restic/restic/internal/backend/logger.(*Backend).Load
	github.com/restic/restic/internal/backend/logger/log.go:46
github.com/restic/restic/internal/backend/retry.(*Backend).Load.func1
	github.com/restic/restic/internal/backend/retry/backend_retry.go:131
github.com/restic/restic/internal/backend/retry.retryNotifyErrorWithSuccess.func1
	github.com/restic/restic/internal/backend/retry/backend_retry.go:47
github.com/cenkalti/backoff/v4.RetryNotifyWithTimer.Operation.withEmptyData.func1
	github.com/cenkalti/backoff/v4@v4.2.1/retry.go:18
github.com/cenkalti/backoff/v4.doRetryNotify[...]
	github.com/cenkalti/backoff/v4@v4.2.1/retry.go:88
github.com/cenkalti/backoff/v4.RetryNotifyWithTimer
	github.com/cenkalti/backoff/v4@v4.2.1/retry.go:61
github.com/cenkalti/backoff/v4.RetryNotify
	github.com/cenkalti/backoff/v4@v4.2.1/retry.go:49
github.com/restic/restic/internal/backend/retry.retryNotifyErrorWithSuccess
	github.com/restic/restic/internal/backend/retry/backend_retry.go:55
github.com/restic/restic/internal/backend/retry.(*Backend).retry
	github.com/restic/restic/internal/backend/retry/backend_retry.go:77
github.com/restic/restic/internal/backend/retry.(*Backend).Load
	github.com/restic/restic/internal/backend/retry/backend_retry.go:129
github.com/restic/restic/internal/backend.LoadAll
	github.com/restic/restic/internal/backend/utils.go:19
github.com/restic/restic/internal/repository.LoadKey
	github.com/restic/restic/internal/repository/key.go:182
github.com/restic/restic/internal/repository.OpenKey
	github.com/restic/restic/internal/repository/key.go:66
github.com/restic/restic/internal/repository.SearchKey.func1
	github.com/restic/restic/internal/repository/key.go:146
github.com/restic/restic/internal/repository.SearchKey.(*Repository).List.func2
	github.com/restic/restic/internal/repository/repository.go:887
github.com/restic/restic/internal/backend/retry.(*Backend).List.func1.1
	github.com/restic/restic/internal/backend/retry/backend_retry.go:178
github.com/restic/restic/internal/backend/rest.(*Backend).listv2
	github.com/restic/restic/internal/backend/rest/rest.go:410
github.com/restic/restic/internal/backend/rest.(*Backend).List
	github.com/restic/restic/internal/backend/rest/rest.go:345
github.com/restic/restic/internal/backend/logger.(*Backend).List
	github.com/restic/restic/internal/backend/logger/log.go:60
github.com/restic/restic/internal/backend/retry.(*Backend).List.func1
	github.com/restic/restic/internal/backend/retry/backend_retry.go:172
github.com/restic/restic/internal/backend/retry.retryNotifyErrorWithSuccess.func1
	github.com/restic/restic/internal/backend/retry/backend_retry.go:47
github.com/cenkalti/backoff/v4.RetryNotifyWithTimer.Operation.withEmptyData.func1
	github.com/cenkalti/backoff/v4@v4.2.1/retry.go:18
github.com/cenkalti/backoff/v4.doRetryNotify[...]
	github.com/cenkalti/backoff/v4@v4.2.1/retry.go:88
github.com/cenkalti/backoff/v4.RetryNotifyWithTimer
	github.com/cenkalti/backoff/v4@v4.2.1/retry.go:61
github.com/cenkalti/backoff/v4.RetryNotify
	github.com/cenkalti/backoff/v4@v4.2.1/retry.go:49
github.com/restic/restic/internal/backend/retry.retryNotifyErrorWithSuccess
	github.com/restic/restic/internal/backend/retry/backend_retry.go:55
github.com/restic/restic/internal/backend/retry.(*Backend).retry
	github.com/restic/restic/internal/backend/retry/backend_retry.go:77
github.com/restic/restic/internal/backend/retry.(*Backend).List
	github.com/restic/restic/internal/backend/retry/backend_retry.go:171
2024/04/04 10:00:29 logger/log.go:47	logger.(*Backend).Load	1	  load err <key/f221949399> does not exist
2024/04/04 10:00:29 repository/key.go:68	repository.OpenKey	1	LoadKey(f221949399db1b03be9a182fb361cedf4e3ea96190f6ec030c8131cfee053220) returned error <key/f221949399> does not exist
2024/04/04 10:00:29 repository/key.go:148	repository.SearchKey.func1	1	key f221949399db1b03be9a182fb361cedf4e3ea96190f6ec030c8131cfee053220 returned error <key/f221949399> does not exist
2024/04/04 10:00:29 logger/log.go:61	logger.(*Backend).List	1	  list err <key/f221949399> does not exist
2024/04/04 10:00:40 restic/cleanup.go:62	main.CleanupHandler	50	signal interrupt received, cleaning up
2024/04/04 10:00:40 restic/global.go:290	main.Warnf	50	[2Ksignal interrupt received, cleaning up
2024/04/04 10:00:40 restic/cleanup.go:87	main.Exit	50	exiting with status code 130


```

Expected behavior
-----------------

restic run successfully without error.

Actual behavior
---------------

restic produces the error listed above.

Do you have any idea what may have caused this?
-----------------------------------------------

I have no idea what the error means, what the reason may be, and how to reproduce the problem. I have no other backend/service/repository to try, so I don't know if the problem is only related to the specific backend/service/repository. Sorry for that :(

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------

restic is the only backup program I have used, and it feels good to me. Thank you for all the hard work!

**Comment by MichaelEischer** [comment]
This looks like a OneDrive problem. The `keys` folder in the repository should contain a key with filename `f221949399db1b03be9a182fb361cedf4e3ea96190f6ec030c8131cfee053220`. However, when restic tries to open that file, then OneDrive was unable to find that file.

rclone usually logs more details in that case, which might be helpful.

**Comment by alicelogos** [comment]
I've discovered that this error may be caused by a bug of rclone (whose fix is planned to be released in rclone v1.67). I regret not having fully tested rclone before reporting here. My apologies :(

I am closing this issue. Thank you for your understanding and support.

**Comment by MichaelEischer** [comment]
> I regret not having fully tested rclone before reporting here. My apologies :(

No problem at all. We're happy to help especially with well written bug reports like yours :-) .

**Comment by willzhang** [comment]
> 我发现这个错误可能是由rclone的一个bug引起的（其修复计划在rclone v1.67中发布）。我很遗憾在此处报告之前没有完全测试 rclone。我很抱歉 ：（
> 
> 我正在关闭这个问题。感谢您的理解和支持。

thanks，same here，resolved by upgrade rclone version to v1.67.0
```
root@ubuntu:~# rclone --version
rclone v1.67.0
```
